### PR TITLE
Add contributor's guide

### PR DESCRIPTION
**Pull request by jbrantly**
First pass at a contributor's guide. Does this make sense? Anything confusing or left out?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+Each test should have an `expectedOutput` directory which contains any webpack
+filesystem output (typically `bundle.js` and possibly `bundle.js.map`) and any 
+console output. stdout should go in `output.txt` and stderr should go in
+`err.txt`.
+
+As a convenience it is possible to regenerate the expected output from the 
+actual output. This is useful when creating new tests and also when making a
+change that affects multiple existing tests. To run use 
+`npm test -- --save-output`. Note that all tests will automatically pass when
```
Comment (basarat) [inline CONTRIBUTING.md]: `-- --save-output`. Typo?

**Comment by basarat** [comment]
nice :sunglasses:

**Comment by johnnyreilly** [comment]
Looks good!

File: CONTRIBUTING.md (hunk 1 above)
Comment (jbrantly) [inline CONTRIBUTING.md]: Not a typo. Allows arguments to be passed to the executing script. See https://docs.npmjs.com/cli/run-script

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+While not absolutely required, it is encouraged that you first open an issue 
+for any bug or feature request. This allows discussion on the proper course of
+action to take before coding begins.
+
+## Building
+
+To build from source you first need to `npm install`. You will also need a
+version of TypeScript available, either globally or install locally with
+`npm install typescript`. Then run `node build`.
```
Comment (blakeembrey) [inline CONTRIBUTING.md]: I'd simply this and make `typescript` a dev dependency. Then add an `npm` script for building the project. Here's my normal set up: https://github.com/blakeembrey/typescript-simple-loader/blob/master/package.json#L15-L21

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+`webpack.config.js` file which follows this general convention:
+
+```javascript
+module.exports = {
+    entry: './app.ts',
+    output: {
+        filename: 'bundle.js'
+    },
+    resolve: {
+        extensions: ['', '.tsx?', '.js']
```
Comment (blakeembrey) [inline CONTRIBUTING.md]: I'm not sure that extensions is a psuedo-regexp - shouldn't it be `'.ts', '.tsx'` as two entries?

File: CONTRIBUTING.md (hunk 2 above)
Comment (jbrantly) [inline CONTRIBUTING.md]: Yea I'm good with changing to `npm run build`. I considered doing that originally. Not sure why I didn't. Regarding the dev dependency, that would probably work as long as it doesn't mess up the [travis build](https://github.com/TypeStrong/ts-loader/blob/master/.travis.yml).

**Comment by blakeembrey** [comment]
Awesome :+1:

File: CONTRIBUTING.md (hunk 3 above)
Comment (jbrantly) [inline CONTRIBUTING.md]: Wow, you're right. I added that on the fly from a config that didn't have the tsx extension and wasn't thinking. Good catch, I'll fix.

**Comment by jbrantly** [comment]
Fixed the extensions issue. The other note regarding the build script I'll do as a separate PR. Thanks everyone.

**Pull request merged**